python_modules/dagster/dagster_tests/asset_defs_tests/test_materialize_to_memory.py

```diff
@@ -115,12 +116,13 @@ def second():
         materialize_to_memory([first, second])
 
     with pytest.raises(
-        DagsterInvalidDefinitionError,
-        match=(
-            "resource with key 'foo' provided to job conflicts with resource provided to assets."
-            " When constructing a job, all resource definitions provided must match by reference"
-            " equality for a given key."
-        ),
+        # TODO: which error is best this or DagsterInvalidDefinitionError?
+        DagsterInvalidInvocationError,
```

Comment:
urgh yea I guess the definition error is a result of the internal processes which create the job, which might be confusing to users. invocation error seems more grokkable to a user encountering this imo